### DISCUSSION TITLE
Running ripgrep gives IO and invalid argument errors

### DISCUSSION THREAD

**Discussion by CrookedJ** [BurntSushi/ripgrep#1688] [General] [answered]
Installed via yum on RHEL 7.5
Not sure if I missed something?

```
$ sudo rg -l '123-?45-?6789' /
/proc/7093/mem: Input/output error (os error 5)
/proc/7093/clear_refs: Invalid argument (os error 22)
/proc/7093/pagemap: Invalid argument (os error 22)
/proc/7095/mem: Input/output error (os error 5)
/proc/7095/clear_refs: Invalid argument (os error 22)
/proc/7095/pagemap: Invalid argument (os error 22)
/proc/7097/mem: Input/output error (os error 5)
/proc/7097/clear_refs: Invalid argument (os error 22)
/proc/7097/pagemap: Invalid argument (os error 22)
/proc/7099/mem: Input/output error (os error 5)
/proc/7099/clear_refs: Invalid argument (os error 22)
/proc/7099/pagemap: Invalid argument (os error 22)
[...]
```

**Comment by BurntSushi** [comment, marked as answer]
Dunno, sorry. Works on my system, and ripgrep is issuing standard syscalls:

    $ rg foobar mem
    mem: Permission denied (os error 13)
    $ sudo rg foobar mem
    $ sudo rg foobar mem --no-mmap
    $ sudo rg foobar mem --mmap
    $

Does the same problem occur with grep?

Also, it is a really bad idea to search `/` with `sudo`. There are issues where people have tried this before. The /proc and /dev directories in particular can be tricky, where even reading a file as root and cause a state change on your system.

**Comment by CrookedJ** [comment]
Oh. That's actually fine then.  
Instead, I'll just search specific directories, I don't necessarily need to search /proc, I just was trying to do an entire file system search in one command.

`$ sudo rg -l '123-?45-?6789' /home` runs fine, I can just run it for all the other directories.

**Reply by BurntSushi** [reply]
@CrookedJ For clarity, does `grep` produce the same error messages when searching `/proc`?

**Reply by CrookedJ** [reply]
Pretty much the same I'd say.

```
$ sudo grep -l -r '123-?45-?6789' /proc
grep: /proc/fs/nfsd/unlock_filesystem: Invalid argument
grep: /proc/fs/nfsd/unlock_ip: Invalid argument
grep: /proc/fs/nfsd/filehandle: Invalid argument
grep: /proc/sys/net/ipv4/route/flush: Permission denied
grep: /proc/sys/net/ipv6/conf/all/stable_secret: Input/output error
grep: /proc/sys/net/ipv6/conf/default/stable_secret: Input/output error
grep: /proc/sys/net/ipv6/conf/enp3s0f0/stable_secret: Input/output error
grep: /proc/sys/net/ipv6/conf/enp3s0f1/stable_secret: Input/output error
^C
```

**Reply by BurntSushi** [reply]
Aye, that's what I would expect. It's definitely not an issue with ripgrep, but I don't know what the difference is between your system and mine. Maybe you have something like SEL enabled? Dunno.

**Reply by CrookedJ** [reply]
Possibly? To be honest, I'm not super familiar with Linux - just trying to complete some tasks for work.
And clearly not familiar enough to realize /proc isn't something I should be searching, heh.

Thank you though
